File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        ans = 0
        l=0
        maxm = 0
        for r in range(len(s)):
            count[s[r]] = 1 + count.get(s[r],0)
            maxm = max(maxm,count[s[r]])
            while (r-l+1) - maxm >k:
                count[s[l]] -= 1
                l += 1
                for i in s:
                    maxm = max(maxm,count.get(i,0))
            ans = max(ans,r-l+1)
        return ans
    

File: Data Structures & Algorithms/merge-intervals/submission-1.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        unordered_map<int,int> m;
        vector<vector<int>> ans;
        int maxm=0;
        for(auto &v:intervals){
            maxm=max(maxm,v[1]);
            m[v[0]]=max(v[1],m[v[0]]);
        }
        for(int i=0;i<=maxm;i++){
            if(!m.count(i)) continue;
            if(ans.empty()){
                ans.push_back({i,m[i]});
                continue;
            }
            if(i<=ans.back()[1]) ans.back()[1]=max(ans.back()[1],m[i]);
            else ans.push_back({i,m[i]});
        }
        return ans;
    }
    
};
